rust.yaml

```diff
@@ -0,0 +1,124 @@
+package:
+  name: rust
+  version: 1.65.0
+  epoch: 0
+  description: "rust a type safe memory safe language"
+  target-architecture:
+    - all
+  copyright:
+    - paths:
+      - "*"
+      attestation: TODO
+      license: Apache-2.0 AND MIT
+  dependencies:
+    runtime:
+    provides: 
+    # hack to allow rust to bootstrap itself
+      - rust-bootstrap=1.65.0
+      
+environment:
+  contents:
+    packages:
+      - busybox
+      - ca-certificates-bundle
+      - build-base
+      - rust-stage0
+      - libgit2-dev
+      - coreutils
+      - curl-dev
+      - file
+      - libssh2-dev
+      - llvm15
+      - libLLVM-15
+      - llvm15-dev
+      - clang-15
+      - openssl-dev
+      - libgit2-dev
+      - libssh2-dev
+      - python3
+      - rust-stage0=>${{package.version}}
+      - tar
+      - wasi-libc
+      - xz-dev
+      - zlib-dev
+
+pipeline:
+  - uses: fetch
+    with:
+      uri: https://static.rust-lang.org/dist/rustc-${{package.version}}-src.tar.xz
+      expected-sha256: 2d6aa2ea3fedd55acbbd65f78349385f9daa9c14e25db7a8df5d015588eee1cf
+      extract: false 
+  - runs: |
+      tar -xJf rustc-${{package.version}}-src.tar.xz
+      rm rustc-${{package.version}}-src.tar.xz
+  - runs: |
+      unset CARGO_PROFILE_RELEASE_LTO
+      unset CARGO_PROFILE_RELEASE_OPT_LEVEL
+      unset CARGO_PROFILE_RELEASE_PANIC
+      unset CARGO_PROFILE_RELEASE_CODEGEN_UNITS
+      export CFLAGS="$CFLAGS -O2"
+      export CXXFLAGS="$CXXFLAGS -O2"
+      export OPENSSL_NO_VENDOR=1
+      export RUST_BACKTRACE=1
+      export ARCH=${{host.triplet.rust}}
+      cd rustc-${{package.version}}-src
+      ./configure \
+        --host="${ARCH}" \
+        --target="${ARCH},wasm32-wasi,wasm32-unknown-unknown" \
+        --prefix="/usr" \
+        --release-channel="stable" \
+        --enable-local-rust \
+        --local-rust-root="/usr" \
+        --llvm-root="/usr/lib/llvm15" \
```

Comment:
use `--llvm-config` flag for specifying llvm-config bin path

https://github.com/rust-lang/rust/blob/2682b88c526d493edeb2d3f2df358f44db69b73f/src/bootstrap/configure.py#L98